boards: riscv: Add support for WeAct Studio ESP32-C3-Mini

heronet opened the pull request:
Add board support for WeAct Studio ESP32-C3-Mini, a compact development board based on ESP32-C3FH4 with 4MB internal flash.

Features:
- ESP32-C3 RISC-V single-core @ 160MHz
- 400KB RAM, 4MB flash
- USB Serial/JTAG for flashing and debugging
- Wi-Fi 802.11b/g/n and Bluetooth 5 LE
- On-board blue LED (GPIO8)
- Boot button (GPIO9)
- Exposed GPIOs: 0-10, 18-19, 20-21

Supported peripherals:
- USB Serial/JTAG
- I2C (GPIO0, GPIO1)
- SPI (GPIO3-6)
- GPIO
- Watchdog
- TRNG
- Wi-Fi
- Bluetooth

sylvioalves commented on boards/weact/esp32c3_mini/weact_esp32c3_mini.yaml:
```diff
@@ -0,0 +1,19 @@
+identifier: weact_esp32c3_mini
+name: ESP32-C6-Mini
```
WeAct Studio ESP32-C3 Mini

sylvioalves commented on boards/weact/esp32c3_mini/board.yml:
```diff
@@ -0,0 +1,6 @@
+board:
+  name: weact_esp32c3_mini
+  full_name: ESP32-C3-Mini
```
Change this to `WeAct Studio ESP32-C3-Mini`.

heronet committed:
boards: riscv: Add support for WeAct Studio ESP32-C3-Mini

Add board support for WeAct Studio ESP32-C3-Mini, a compact
development board based on ESP32-C3FH4 with 4MB internal flash.

Features:
- ESP32-C3 RISC-V single-core @ 160MHz
- 400KB RAM, 4MB flash
- USB Serial/JTAG for flashing and debugging
- Wi-Fi 802.11b/g/n and Bluetooth 5 LE
- On-board blue LED (GPIO8)
- Boot button (GPIO9)
- Exposed GPIOs: 0-10, 18-19, 20-21

Supported peripherals:
- USB Serial/JTAG
- I2C (GPIO0, GPIO1)
- SPI (GPIO3-6)
- GPIO
- Watchdog
- TRNG
- Wi-Fi
- Bluetooth

Signed-off-by: Siratul Islam <sirat4757@gmail.com>
